src/main/java/me/jellysquid/mods/sodium/client/render/occlusion/BlockOcclusionCache.java

```diff
@@ -35,7 +35,7 @@ public boolean shouldDrawSide(BlockState selfState, BlockView view, BlockPos pos
 
         BlockState adjState = view.getBlockState(adjPos);
 
-        if (selfState.isSideInvisible(adjState, facing)) {
+        if (selfState.isSideInvisible(adjState, facing) || selfState.supportsExternalFaceHiding() && selfState.hidesNeighborFace(view, pos, adjState, facing.getOpposite())) {
```

Comment:
You should swap these two calls: The default implementation of `hidesNeighborFace` is just `return false`, while the default impl of `supportsExternalFaceHiding` has a call to `ForgeHooksClient::isBlockInSolidLayer`, which involves multiple map lookups. With this order you always have the map lookups, while the other order will virtually never do them due to short-circuiting. I've seen profiles where the call to `supportsExternalFaceHiding` accounts for 10% of chunk building time, and that's without Sodium's code making the rest of it faster.

Also, the `hidesNeighborFace` call needs to be on `adjState` with `selfState` as a parameter rather than the other way around.